src/lsm/README.md

```diff
@@ -0,0 +1,111 @@
+# Tree
+## Compaction
+
+Tree compaction runs to the sound of music!
+
+Compacting LSM trees involves merging and moving tables into the next levels as needed.
+To avoid write amplification stalls and bound latency, compaction is done incrementally.
+
+A full compaction phase is denoted as a bar or measure, using terms from music notation.
+Each bar consists of `lsm_batch_multiple` beats or "compaction ticks" of work.
+A compaction beat is started asynchronously with `compact()` which takes a callback and
+the op that was just committed.
+
+A bar is split in half according to the "first" down beat and "middle" down beat.
```

Comment:
What's a down beat? What order do down/up beats happen in?

Reply:
I assume this is part of the music analogy :man_shrugging:. It is spelled out more explicity down below, lines 31-49.

Reply:
You could probably just remove `down` from this sentence. First beat and middle beat are pretty clear, just the 'down' that is confusing as not introduced yet.